subsys/net/lib/lwm2m/lwm2m_shell.c

```diff
@@ -71,9 +89,192 @@ static int cmd_send(const struct shell *sh, size_t argc, char **argv)
 	return 0;
 }
 
+static int cmd_exec(const struct shell *sh, size_t argc, char **argv) 
+{
+	struct lwm2m_ctx *ctx = lwm2m_rd_client_ctx();
+
+	if (!ctx) {
+		shell_error(sh, "no lwm2m context yet\n");
+		return -ENOEXEC;
+	}
+
+	int ignore_cnt = 2;	/* Subcmd + PATH */
+	const char* pathstr = argv[1];
+	struct lwm2m_engine_res *res = NULL;
+	int ret = lwm2m_engine_get_resource(pathstr, &res);
+	if (ret < 0)
+	{
+		shell_error(sh, "Couldn't execute resource\n");
+		return ret;
+	}
+	if(res->execute_cb)
+	{
+		uint16_t obj_id;
+		sscanf(pathstr, "%*d/%hd/%*d", &obj_id);
+		return res->execute_cb(obj_id, argv[ignore_cnt], argc-ignore_cnt);
+	}
+	return 0;
```

Comment:
Now you don't print out anything, if the resource does not have execute callback.
Also, we cannot return what the callback returned, because from shell point of view, the execution was success. So maybe print it out, if the return was not zero.
```suggestion
	if(!res->execute_cb) {
	        shell_error(sh, "No execute callback\n");
	        return -EINVAL;
	}        
	uint16_t obj_id;

	sscanf(pathstr, "%*d/%hd/%*d", &obj_id);
	ret = res->execute_cb(obj_id, argv[ignore_cnt], argc-ignore_cnt);
	if (ret) {
	        shell_error(sh, "returned %d\n", ret);
	        return -ENOEXEC;
	}
	return 0;
```